### PR TITLE
Add "trackAllPages" and "excludeUris" settings. The first allows acti…

### PR DESCRIPTION
**Pull request by kgrruz**
…vating or not the automatic tracking through the 'post_controller_constructor' event and the second excluding certain uri's from tracking.

### CODE REVIEW

**Comment by MGatner** [comment]
I'm thrilled to have a PR here, thanks for contributing! Just a heads up that in the next major version this library will be refactored into a Filter, which will allow developers to apply/exclude it to specific routes. 
If you would still like to proceed with your feature submission please run CS Fixer (`composer style`) to enforce the style guide. Normally I'd want tests but this library is very old and I never wrote tests for it so I won't enforce that.

File: src/Config/Events.php [hunk 1]
```diff
@@ -3,7 +3,11 @@
 use CodeIgniter\Events\Events;
 use Config\Services;
 
-Events::on('post_controller_constructor', function () {
+Events::on('post_controller_constructor', function () { 	
+	
+	$config = config('Tatter\Visits\Config\Visits');
```
Comment (MGatner) [inline src/Config/Events.php]: Shorthand Config names are preferable:
```suggestion
	$config = config('Visits');
```

File: src/Config/Visits.php [hunk 2]
```diff
@@ -4,6 +4,10 @@
 
 class Visits extends BaseConfig
 {
+	
+	//Enable tracking in all controllers using the post_controller_constructor event
+	public $trackAllPages = false;
```
Comment (MGatner) [inline src/Config/Visits.php]: This needs to be `true` by default or current implementations will stop tracking when they upgrade. 
```suggestion
	public $trackAllPages = true;
```

File: src/Visits.php [hunk 3]
```diff
@@ -109,13 +109,24 @@ public function record()
...
+		
 		// start the object with parsed URL components (https://secure.php.net/manual/en/function.parse-url.php)
 		$visit->fill(parse_url( current_url() ));
 
+		// Check if URI has been whitelisted from Visit check
+		if ($exclude_uris = $this->config->excludeUris)
+		{ 
+			foreach ($exclude_uris as $excluded)
+			{ 
+				if (preg_match('#^'.$excluded.'$#iu', $visit->path))
```
Comment (MGatner) [inline src/Visits.php]: Better to use CodeIgniter's helper function:
```suggestion
				if (url_is($excluded))
```

https://codeigniter4.github.io/CodeIgniter4/helpers/url_helper.html#url_is

**Review by MGatner — CHANGES_REQUESTED** [review]

**Comment by MGatner** [comment]
@kgrruz I'm sorry, I shouldn't have told you to run CS Fixer as part of this PR - I didn't realize this repo hadn't been updated to use the CodeIgniter Style Guide. I will resolve the style and actions in the other PR and then see about cherry-picking your changes. Hold tight.

**Comment by kgrruz** [comment]
OK. Thanks for the suggestions.

**Pull request closed without merging**